src/boot/bootctl-random-seed.c

```diff
@@ -17,6 +18,39 @@
 #include "tmpfile-util.h"
 #include "umask-util.h"
 
+static int random_seed_verify_permissions(int fd, mode_t expected_type) {
+        _cleanup_free_ char *full_path = NULL;
+        struct stat st;
+        int r;
+
+        assert(fd >= 0);
+
+        r = fd_get_path(fd, &full_path);
+        if (r < 0)
+                return log_error_errno(r, "Unable to determine full path of random seed fd: %m");
+
+        if (fstat(fd, &st) < 0)
+                return log_error_errno(errno, "Unable to stat %s: %m", full_path);
+
+        if (((st.st_mode ^ expected_type) & S_IFMT) != 0)
```

Comment:
Looks reasonable.
Isn't this the same as `(st.st_mode & S_IFMT) != expected_type` ?

Reply:
yes

Reply:
well almost, i.e. same as `(st.st_mode & S_IFMT) != (expected_type & S_IFMT)`

just shorter